src/Rules/Classes/ImpossibleInstanceOfRule.php

```diff
@@ -57,6 +60,20 @@ public function processNode(Node $node, Scope $scope): array
 			}
 		}
 
+		if ($className !== null) {
+			$classReflection = $this->reflectionProvider->getClass($className);
```

Comment:
You first need to ask about hasClass, only then it's safe to call getClass.

Reply:
ah of course, thanks